builtins: implement `to_char` for timestamp types

otan opened the pull request:
Informs #91156

Release note (sql change): The `to_char(timestamp, string)` is now implemented, except for the `rm` format directive.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/91380)
<!-- Reviewable:end -->

otan commented:
backporting this will unfortunately be manual :'( the builtins.go has some mad changes

otan committed:
builtins: implement `to_char` for timestamp types

Release note (sql change): The `to_char(timestamp, string)` is now
implemented, except for the `rm` format directive.

otan commented:
have a bigger PR coming for this, closing for now!
